### DISCUSSION TITLE
How do I use the command line(cli) to pass arguments?

### DISCUSSION THREAD

**Discussion by imporoutco586** [dagster-io/dagster#21076] [Q&A]
Hi,

I want to know if I can use cli to pass parameters when I run some assets or jobs. I didn't find anything similarly relevant in the dagster aseet command, but I did find the -c config command in dagster job. This is my code:

```
@asset
def trainer(num_workers=2):
    trainer = TorchTrainer(
        train_func_distributed,
        scaling_config=ScalingConfig(num_workers=num_workers, use_gpu=use_gpu)
    )

    results = trainer.fit()
    return results

trainer_job = define_asset_job(name="trainer_job", selection="trainer")
```

For example, if I want to change num_workers when I use `dagster job execute -c config.yaml` command， how can I write in the config.yaml or use some other command directly? I'm not sure if this asset-derived job can be done using the -c config method, so please let me know! If there is a direct method to pass parameters to the asset also welcome to inform!

Any answer will be greatly appreciated!

**Comment by yuhan** [comment]
You can use the CLI to pass parameters when you run asset-based jobs in Dagster. When executing a job from Dagster's CLI with the `dagster job execute` command, you can put the configuration in a YAML file and then pass the file path with the `--config` option.

For your specific case, you would define the configuration for the `trainer` asset in a YAML file. Here's an example of how you might write your `config.yaml` to pass the `num_workers` parameter:

```yaml
ops:
  trainer:
    config:
      num_workers: 4
```

Then you would execute your job with the following command:

```shell
dagster job execute --config config.yaml
```
